### ISSUE TITLE
image: PATH should work

### ISSUE THREAD

**Issue by ibuildthecloud** [rancher/rio#432]
In a Riofile the following should work

```yaml
services:
  foo:
    image: ./blah
```

In this situation we should detect that ./blah is a path and then setup the `build` element appropriately.

**Comment by daxmc99** [comment]
Should this work for remote repositories and local ones?

**Comment by daxmc99** [comment]
This can be tested with https://github.com/daxmc99/rio-demo/tree/test2
Clone and run `rio up` in this directory

**Comment by izaac** [comment]
@daxmc99 in `rio v0.6.0` I end up with the following after `rio up`:

```bash
➜  dax-rio-demo git:(test2) rio ps
NAME      IMAGE                                                                                         ENDPOINT   SCALE     APP       VERSION    WEIGHT    CREATED          DETAIL
nginx     default/nginx-nginx@sha256:56a6cd38367f6dac4cbf88e8fd60ee3be77fd34aa5e7b6a925e4c6cf9a38e6dc              0/1 0%    nginx     60fe91b3   100%      44 seconds ago   nginx: nginx(ErrImagePull/rpc error: code = Unknown desc = failed to pull and unpack image "localhost:5442/default/nginx-nginx@sha256:56a6cd38367f6dac4cbf88e8fd60ee3be77fd34aa5e7b6a925e4c6cf9a38e6dc": failed to resolve reference "localhost:5442/default/nginx-nginx@sha256:56a6cd38367f6dac4cbf88e8fd60ee3be77fd34aa5e7b6a925e4c6cf9a38e6dc": failed to do request: Head https://localhost:5442/v2/default/nginx-nginx/manifests/sha256:56a6cd38367f6dac4cbf88e8fd60ee3be77fd34aa5e7b6a925e4c6cf9a38e6dc: http: server gave HTTP response to HTTPS client)
```

This is a k3s v0.10.0 cluster hosted on Linode.
```bash 
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2-k3s.1", GitCommit:"b8b17ba55f20e590df507fce333dfee13ab438c6", GitTreeState:"clean", BuildDate:"2019-10-16T05:17Z", GoVersion:"go1.13.3", Compiler:"gc", Platform:"linux/amd64"}```

**Comment by rancher-max** [comment]
Looks like this is still failing in v0.6.0-alpha.2.

Ran `rio up --name dax --branch test2 https://github.com/daxmc99/rio-demo`
The build is stuck in `Running`. Build logs:
```
 rio logs -a taskrun/default/default-dax-stack-629c5-dade0
+ default default-dax-stack-629c5-dade0-pod-ef37fa › step-credential-initializer-p94jd
+ default default-dax-stack-629c5-dade0-pod-ef37fa › step-place-tools
+ default default-dax-stack-629c5-dade0-pod-ef37fa › step-working-dir-initializer-5h4jt
+ default default-dax-stack-629c5-dade0-pod-ef37fa › step-git-source-source-4gtcd
default-dax-stack-629c5-dade0-pod-ef37fa step-working-dir-initializer-5h4jt {"level":"warn","ts":1572558866.39665,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: open /var/run/ko/HEAD: no such file or directory"}
default-dax-stack-629c5-dade0-pod-ef37fa step-working-dir-initializer-5h4jt {"level":"info","ts":1572558866.3982472,"logger":"fallback-logger","caller":"bash/main.go:64","msg":"Successfully executed command \"sh -c mkdir -p /workspace/source\"; output "}
default-dax-stack-629c5-dade0-pod-ef37fa step-credential-initializer-p94jd {"level":"warn","ts":1572558865.0499465,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/master\" is not a valid GitHub commit ID"}
default-dax-stack-629c5-dade0-pod-ef37fa step-credential-initializer-p94jd {"level":"info","ts":1572558865.05034,"logger":"fallback-logger","caller":"creds-init/main.go:40","msg":"Credentials initialized."}
default-dax-stack-629c5-dade0-pod-ef37fa step-git-source-source-4gtcd {"level":"warn","ts":1572558871.501459,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/master\" is not a valid GitHub commit ID"}
default-dax-stack-629c5-dade0-pod-ef37fa step-git-source-source-4gtcd {"level":"info","ts":1572558871.966079,"logger":"fallback-logger","caller":"git/git.go:103","msg":"Successfully cloned https://github.com/daxmc99/rio-demo @ 54463a937703b70ed277d250dd574a2831b0ee1d in path /workspace/source"}
```

**Comment by izaac** [comment]
@StrongMonkey in my case  I'm seeing this error message:

```bash
default-dax-stack-629c5-dade0-pod-a80549 step-rio-up time="2019-11-04T16:45:13Z" level=error msg="pods \"buildkitd-5bf7bcb755-4kl49\" is forbidden: User \"system:serviceaccount:default:default-dax-stack-629c5-dade0-stack\" cannot get resource \"pods\" in API group \"\" in the namespace \"rio-system\""
```
Looks like some privileges issue?

**Comment by StrongMonkey** [comment]
@izaac Correct, i fixed that in the referenced PR. I will move this to to-test once new alpha is out.

**Comment by StrongMonkey** [comment]
Available to test in v0.6.0-alpha3. @izaac

**Comment by rancher-max** [comment]
This works for me if I follow this instructions here: https://github.com/rancher/rio/issues/432#issuecomment-533735908

However, it fails with this command: `rio up --name dax --branch test2 https://github.com/daxmc99/rio-demo` which I would expect to work.

**Comment by rancher-max** [comment]
I believe my previous comment was due to an incorrect setup. I changed the Riofile to:
```
services:
  mango:
    image: ./blah/
    ports:
    - "8080"
```
in a repo that had `./blah/Dockerfile` and it succeeded.

Closing as fixed in v0.6.0-alpha.3
